pkg/query/iter.go

```diff
@@ -153,33 +184,47 @@ func (s *chunkSeries) Iterator() storage.SeriesIterator {
 	var sit storage.SeriesIterator
 	its := make([]chunkenc.Iterator, 0, len(s.chunks))
 
-	switch s.aggr {
-	case resAggrCount:
-		for _, c := range s.chunks {
-			its = append(its, getFirstIterator(c.Count, c.Raw))
-		}
-		sit = newChunkSeriesIterator(its)
-	case resAggrSum:
-		for _, c := range s.chunks {
-			its = append(its, getFirstIterator(c.Sum, c.Raw))
-		}
-		sit = newChunkSeriesIterator(its)
-	case resAggrMin:
-		for _, c := range s.chunks {
-			its = append(its, getFirstIterator(c.Min, c.Raw))
-		}
-		sit = newChunkSeriesIterator(its)
-	case resAggrMax:
-		for _, c := range s.chunks {
-			its = append(its, getFirstIterator(c.Max, c.Raw))
-		}
-		sit = newChunkSeriesIterator(its)
-	case resAggrCounter:
-		for _, c := range s.chunks {
-			its = append(its, getFirstIterator(c.Counter, c.Raw))
+	if len(s.aggrs) == 1 {
+		switch s.aggrs[0] {
+		case storepb.Aggr_COUNT:
+			for _, c := range s.chunks {
+				its = append(its, getFirstIterator(c.Count, c.Raw))
+			}
+			sit = newChunkSeriesIterator(its)
+		case storepb.Aggr_SUM:
+			for _, c := range s.chunks {
+				its = append(its, getFirstIterator(c.Sum, c.Raw))
+			}
+			sit = newChunkSeriesIterator(its)
+		case storepb.Aggr_MIN:
+			for _, c := range s.chunks {
+				its = append(its, getFirstIterator(c.Min, c.Raw))
+			}
+			sit = newChunkSeriesIterator(its)
+		case storepb.Aggr_MAX:
+			for _, c := range s.chunks {
+				its = append(its, getFirstIterator(c.Max, c.Raw))
+			}
+			sit = newChunkSeriesIterator(its)
+		case storepb.Aggr_COUNTER:
+			for _, c := range s.chunks {
+				its = append(its, getFirstIterator(c.Counter, c.Raw))
+			}
+			sit = downsample.NewCounterSeriesIterator(its...)
+		default:
+			return errSeriesIterator{err: errors.Errorf("unexpected result aggregate type %v", s.aggrs)}
 		}
-		sit = downsample.NewCounterSeriesIterator(its...)
-	case resAggrAvg:
+		return newBoundedSeriesIterator(sit, s.mint, s.maxt)
+	}
+
+	if len(s.aggrs) != 2 {
+		return errSeriesIterator{err: errors.Errorf("unexpected result aggregate type %v", s.aggrs)}
+	}
+
+	switch {
+	case s.aggrs[0] == storepb.Aggr_SUM && s.aggrs[1] == storepb.Aggr_COUNT,
```

Comment:
When is this case possible? `aggrsFromFunc` only returns the 2nd case.

Reply:
This is average

Reply:
ah I know what you mean. You are right but it's easy to swap and assume it will handled well, so I would vote to be extra safe here.